startup_assistance/indico_startup_assistance/migrations/20190220_1138_b160812c186b_add_startup_assistance_requests_table.py

```diff
@@ -0,0 +1,68 @@
+"""Add startup assistance requests table
+
+Revision ID: b160812c186b
+Revises:
+Create Date: 2019-02-20 11:38:08.691528
+"""
+
+from __future__ import unicode_literals
+
+import sqlalchemy as sa
+from alembic import context, op
+from sqlalchemy.sql.ddl import CreateSchema, DropSchema
+
+
+# revision identifiers, used by Alembic.
+revision = 'b160812c186b'
+down_revision = None
+branch_labels = None
+depends_on = None
+
+
+def upgrade():
+    if context.is_offline_mode():
+        raise Exception('This upgrade is only possible in online mode')
+
+    op.execute(CreateSchema('plugin_startup_assistance'))
+    op.create_table(
+        'startup_assistance_requests',
+        sa.Column('reservation_id', sa.Integer(), nullable=False),
+        sa.ForeignKeyConstraint(['reservation_id'], ['roombooking.reservations.id']),
+        sa.PrimaryKeyConstraint('reservation_id'),
+        schema='plugin_startup_assistance'
+    )
+
+    conn = op.get_bind()
+    res = conn.execute('''
+        SELECT id FROM roombooking.reservations
+        WHERE needs_assistance
+        AND end_dt > NOW()::TIMESTAMP
+        AND state IN (1, 2)
+    ''')
+    for row in res:
+        conn.execute('''
+            INSERT INTO plugin_startup_assistance.startup_assistance_requests
+            VALUES (%s)
+        ''', (row.id,))
+
+    op.drop_column('reservations', 'needs_assistance', schema='roombooking')
+
+
+def downgrade():
+    if context.is_offline_mode():
+        raise Exception('This downgrade is only possible in online mode')
+
+    op.add_column('reservations',
+                  sa.Column('needs_assistance', sa.Boolean(), nullable=False, server_default='false'),
+                  schema='roombooking')
+
+    conn = op.get_bind()
+    res = conn.execute('SELECT reservation_id FROM plugin_startup_assistance.startup_assistance_requests')
+    for row in res:
```

Comment:
just assume the column in reservation exists and then do something like `UPDATE reservations SET ... = true WHERE EXISTS (...)` in a single query